src/ase/cw/control/OrderController.java

```diff
@@ -115,6 +117,13 @@ public void addRandomOrder(boolean priority) throws InvalidCustomerIdException {
     }
 
     public void addServer() {
+        synchronized (serverList){
+            if(!applicationClosing){ // If the application is currently closing(if the factory is done and the queue is empty) it is not allowed to add new servers
+                currentServerNumber=this.serverList.size()+1;
+                this.serverList.add(new ServerController(currentServerNumber,queueFrame,queuedOrders,this));
+            }
+        }
+/*
```

Comment:
this is not right. if you have servers 1,2,3 and then you close 2 the next server you add will be 3 so you will have 1,3,3. I added this as a fix in my last PR so please use that